packages/forms/src/elements/common/Hint.tsx

```diff
@@ -18,6 +18,21 @@ export const Hint = React.forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivEleme
     const fieldContext = useFieldContext();
     const type = useInputContext();
 
+    useEffect(() => {
+      if (fieldContext) {
+        fieldContext.setHint(true);
+      }
+
+      return () => {
+        if (fieldContext) {
+          fieldContext.setHint(false);
+        }
+      };
+      // since we only want to run this once, if context exists,
+      // we use an empty dependency array
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
@jzempel Why does our eslint configuration flag an empty array as it is a valid case in a component's lifecycle?